centinel/primitives/http.py

```diff
@@ -44,5 +47,80 @@ def get_request(host, path="/", headers=None, ssl=False):
         "response" : response,
         "request"  : request
     }
-
+    # the external result is used when threading to store
+    # the results in the list container provided.
+    if external is not None and type(external) is dict:
+        external[url] = result
     return result
+
+def get_requests_batch(input_list, delay_time=0.5, max_threads=100):
+    """
+    This is a parallel version of the HTTP GET primitive.
+
+    Params:
+    input_list- the input is a list of either dictionaries containing
+                query information, or just domain names (and NOT URLs).
+    delay_time- delay before starting each thread
+    max_threads- maximum number of concurrent threads
+
+    Note: the input list can look like this:
+    [
+        { "host": "www.google.com",   "path": "/", "headers": [],
+          "ssl": False, "url": "http://www.google.com/" },
+        "www.twitter.com",
+        "www.youtube.com",
+        { "host": "www.facebook.com", "path": "/", "headers": [],
+          "ssl": True, "url": "http://www.facebook.com" },
+        ...
+    ]
+
+    """
+    results = {}
+    threads = []
+    for row in input_list:
+        headers = []
+        path = "/"
+        ssl = False
+        theme = "http"
+        if type(row) is dict:
+            if "host" not in row:
+                continue
+            host = row["host"]
+
+            if "path" in row:
+                path = row["path"]
+
+            if "headers" in row:
+                if type(row["headers"]) is list:
+                    headers = row["headers"]
+
+            if "ssl" in row:
+                ssl = row["ssl"]
+                theme = "https"
+
+            if "url" in row:
+                url = row["url"]
+            else:
+                url = "%s://%s%s" % (theme, host, path)
+        else:
+            host = row
+            url = "%s://%s%s" % (theme, host, path)
+
+
+        while threading.active_count() > max_threads:
+            time.sleep(1)
```

Comment:
> what if we have a bunch of threads that hang?

A "bunch" of them? This will only loop if **all** 100 concurrent threads hang together all at the same time.